projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts

```diff
@@ -1098,9 +1098,11 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
     private _updateVScrollOffset() {
         let scrollOffset = 0;
         const vScroll = this.vh.instance.elementRef.nativeElement;
-        scrollOffset = vScroll && parseInt(vScroll.style.height, 10) ?
-        vScroll.scrollTop - this.sizesCache[this.state.startIndex] : 0;
-        this.dc.instance._viewContainer.element.nativeElement.style.top = -(scrollOffset) + 'px';
+        if (vScroll) {
+            scrollOffset = this.fixedUpdateAllRows(this._virtScrollTop);
```

Comment:
@nrobakova I don't think it is necessary to trigger a full update on all rows here. The problem seems to be that since the igxForOf data is changed (due to page change), the sizesCache is updated but the startIndex is not. Updating the startIndex on _applyChanges before the embedded views are updated should fix this.